python/lsst/sconsUtils/state.py

```diff
@@ -172,14 +172,20 @@ def _initEnvironment():
     if env['PLATFORM'] == 'darwin':
         env['LDMODULESUFFIX'] = ".so"
         if not re.search(r"-install_name", str(env['SHLINKFLAGS'])):
-            env.Append(SHLINKFLAGS=["-Wl,-install_name", "-Wl,${TARGET.file}"])
+            env.Append(SHLINKFLAGS=["-install_name", "@rpath/${TARGET.file}"])
```

Comment:
Not obvious if `@rpath/libfoo.dylib` is the right solution here. I think this implies we're putting all our libraries into some joint `lib` (or that we'll be adding a lot of RPATHs into the executables, for each per-product lib)?

It don't think it matters with the current `DYLD_...` EUPS scheme. But will make it impossible for `conda-build` to infer the proper `install_name` from `$PREFIX`, which may make it more difficult to do an EUPS-less install.

I'd propose to leave it as-is until we have a firmer grasp about what's the right thing to do (probably as a part of developing an EUPS-less install?).